compiler/ras/Tree.cpp

```diff
@@ -1585,7 +1585,9 @@ TR_Debug::printNodeInfo(TR::Node * node, TR_PrettyPrinterString& output, bool pr
 
    if (node->getOpCode().isNullCheck())
       {
-      output.append(" on %s%dn", globalIndexPrefix.getStr(), node->getNullCheckReference()->getGlobalIndex());
+      if (node->getNullCheckReference())
+         output.append(" on %s%dn", globalIndexPrefix.getStr(), node->getNullCheckReference()->getGlobalIndex());
+      else output.append(" on null NullCheckReference ----- INVALID tree!!\n");
```

Comment:
From a consistency POV, this message ends with a newline but all others in a similar context don't.  Arguably, trying to print an invalid tree is breaking the mold, but can you confirm that the newline is what you want here?